jax/_src/numpy/lax_numpy.py

```diff
@@ -3780,9 +3780,6 @@ def take(a, indices, axis=None, out=None, mode=None):
 
 def _normalize_index(index, axis_size):
   """Normalizes an index value in the range [-N, N) to the range [0, N)."""
-  if type(axis_size) is Poly:
```

Comment:
This code does not work if `index` is a tracer, and it seems like a partial attempt to make this work for polymorphism. Nothing breaks if I remove this. I am curious if something breaks in Flax.